packages/experience-builder-sdk/src/hooks/useMediaQuery.ts

```diff
@@ -0,0 +1,179 @@
+  toCSSAttribute,
+  toCSSString,
+  toMediaQuery,
+} from '@contentful/experiences-core';
+import { EMPTY_CONTAINER_HEIGHT } from '@contentful/experiences-core/constants';
+import { Breakpoint, ComponentTreeNode } from '@contentful/experiences-core/types';
+import md5 from 'md5';
+import { useMemo } from 'react';
+
+type ResolvedStylesheetData = Array<{
+  className: string;
+  breakpointCondition: string;
+  css: string;
+}>;
+
+export const resolveClassNamesFromBuiltInStyles = ({
+  designPropsByBreakpointId,
+  breakpoints,
+  node,
+}: {
+  designPropsByBreakpointId: Record<string, Record<string, any>>;
+  breakpoints: Breakpoint[];
+  node: ComponentTreeNode;
+}): ResolvedStylesheetData => {
+  const mapOfDesignVariableKeys = flattenDesignTokenRegistry(designTokensRegistry);
+  const currentNodeClassNames: string[] = [];
+
+  const result: Array<{
+    className: string;
+    breakpointCondition: string;
+    css: string;
+  }> = [];
+
+  // then for each breakpoint
+  for (const breakpoint of breakpoints) {
+    const designProps = designPropsByBreakpointId[breakpoint.id];
+    if (!designProps) {
+      continue;
+    }
+
+    const propsByBreakpointWithResolvedDesignTokens = Object.entries(designProps).reduce(
+      (acc, [propName, propValue]) => {
+        return {
+          ...acc,
+          [propName]: maybePopulateDesignTokenValue(propName, propValue, mapOfDesignVariableKeys),
+        };
+      },
+      {},
+    );
+
+    // We convert cryptic prop keys to css variables
+    // Eg: cfMargin to margin
+    const stylesForBreakpoint = buildCfStyles(propsByBreakpointWithResolvedDesignTokens);
+
+    if (
+      !node.children.length &&
+      isStructureWithRelativeHeight(node.definitionId, stylesForBreakpoint.height)
+    ) {
+      stylesForBreakpoint.minHeight = EMPTY_CONTAINER_HEIGHT;
+    }
+
+    const stylesForBreakpointWithoutUndefined: Record<string, string> = Object.fromEntries(
+      Object.entries(stylesForBreakpoint)
+        .filter(([, value]) => value !== undefined)
+        .map(([key, value]) => [toCSSAttribute(key), value]),
+    );
+
+    /**
+         * stylesForBreakpoint {
+            margin: '0 0 0 0',
+            padding: '0 0 0 0',
+            'background-color': 'rgba(246, 246, 246, 1)',
+            width: '100%',
+            height: 'fit-content',
+            'max-width': 'none',
+            border: '0px solid rgba(0, 0, 0, 0)',
+            'border-radius': '0px',
+            gap: '0px 0px',
+            'align-items': 'center',
+            'justify-content': 'safe center',
+            'flex-direction': 'column',
+            'flex-wrap': 'nowrap',
+            'font-style': 'normal',
+            'text-decoration': 'none',
+            'box-sizing': 'border-box'
+          }
+        */
+    // I create a hash of the object above because that would ensure hash stability
+    const styleHash = md5(`${node.id}-${JSON.stringify(stylesForBreakpointWithoutUndefined)}`);
```

Comment:
[optional, note for later]
Given the insights from [#993](https://github.com/contentful/experience-builder/pull/993), we could use here the breakpoint ID as well. This allows us to still reuse class names across nodes that are completely similar instead of creating one class for each node and each breakpoint for the whole tree.
Just a micro-optimization so we can definitely go ahead and improve this later.

```suggestion
    const styleHash = md5(`${breakpoint.id}-${JSON.stringify(stylesForBreakpointWithoutUndefined)}`);
```

Reply:
Initially Daniel was using the breakpoint, but it wasn't working and then he switched to nodeId. We'd need to test both cases to confirm if that's still the case and to understand why this happens.